src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/SortRewriter.cs

```diff
@@ -47,21 +47,30 @@ public override Expression VisitSqlRoot(SqlRootExpression expression, SqlSearchO
             // If the parameter being sorted on is part of a filter then we don't need to run the seperate search for resources that are missing a value for the field being sorted on.
             // If the parameter being sorted on is not part of a filter we need to run a seperate search to get resources that don't have a value for the field being sorted on.
             bool matchFound = false;
+            bool isSortHasMissingModifier = false;
             for (int i = 0; i < expression.SearchParamTableExpressions.Count; i++)
             {
                 Expression updatedExpression = expression.SearchParamTableExpressions[i].Predicate.AcceptVisitor(this, context);
+
+                if (expression.SearchParamTableExpressions[i].Predicate is MissingSearchParameterExpression)
+                {
+                    MissingSearchParameterExpression misingSPExpression = (MissingSearchParameterExpression)expression.SearchParamTableExpressions[i].Predicate;
+                    isSortHasMissingModifier = !misingSPExpression.IsMissing && context.Sort[0].searchParameterInfo.Name.Equals(misingSPExpression.Parameter.Name, System.StringComparison.OrdinalIgnoreCase);
```

Comment:
This should have a check so that if it is ever set to true it can't be set to false if a later search param doesn't meet the criteria.

Reply:
Fixed it.